lib/os/json.c

```diff
@@ -390,6 +390,42 @@ static int arr_next(struct json_obj *json, struct json_token *value)
 	return element_token(value->type);
 }
 
+static void skip_field(struct json_obj *obj,
+		       struct json_obj_key_value *kv)
+{
+	int i = 1;
+
+	if (kv->value.type == JSON_TOK_ARRAY_START) {
+		while (arr_next(obj, &kv->value) >= 0) {
```

Comment:
Good suggestion. Will change to this (because the evaluation order has to be the other way round):
```c
while (i > 0 && arr_next(obj, &kv->value) >= 0) {
```